packages/desktop-client/src/components/budget/MonthPicker.js

```diff
@@ -0,0 +1,166 @@
+  const [targetMonthCount, setTargetMonthCount] = useState(12);
+
+  const currentMonth = monthUtils.currentMonth();
+  const firstSelectedMonth = startMonth;
+
+  const lastSelectedMonth = monthUtils.addMonths(
+    firstSelectedMonth,
+    numDisplayed - 1,
+  );
+
+  const range = monthUtils.rangeInclusive(
+    monthUtils.subMonths(
+      firstSelectedMonth,
+      targetMonthCount / 2 - numDisplayed / 2,
+    ),
+    monthUtils.addMonths(
+      lastSelectedMonth,
+      targetMonthCount / 2 - numDisplayed / 2,
+    ),
+  );
+
+  const firstSelectedIndex =
+    Math.floor(range.length / 2) - Math.floor(numDisplayed / 2);
+  const lastSelectedIndex = firstSelectedIndex + numDisplayed - 1;
+
+  const [size, setSize] = useState('small');
+  const containerRef = useResizeObserver(rect => {
+    setSize(rect.width <= 400 ? 'small' : 'big');
+    setTargetMonthCount(
+      Math.min(Math.max(Math.floor(rect.width / 50), 12), 24),
+    );
+  });
+
+  let yearHeadersShown = [];
+
+  return (
+    <View
+      style={[
+        {
+          flexDirection: 'row',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+        },
+        style,
+      ]}
+    >
+      <View
+        innerRef={containerRef}
+        style={{
+          flexDirection: 'row',
+          flex: 1,
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+      >
+        {range.map((month, idx) => {
+          const monthName = monthUtils.format(month, 'MMM');
+          const selected =
+            idx >= firstSelectedIndex && idx <= lastSelectedIndex;
+
+          const lastHoverId = hoverId + numDisplayed - 1;
+          const hovered =
+            hoverId === null ? false : idx >= hoverId && idx <= lastHoverId;
+
+          const current = currentMonth === month;
+          const year = monthUtils.getYear(month);
+
+          let showYearHeader = false;
+
+          if (!yearHeadersShown.includes(year)) {
+            yearHeadersShown.push(year);
+            showYearHeader = true;
+          }
+
+          const isMonthBudgeted =
+            month >= monthBounds.start && month <= monthBounds.end;
+
+          return (
+            <View
+              key={month}
+              style={[
+                {
+                  padding: '3px 3px',
+                  width: size === 'big' ? '35px' : '20px',
+                  textAlign: 'center',
+                  userSelect: 'none',
+                  cursor: 'default',
+                  borderRadius: 2,
+                  border: 'none',
+                },
+                !isMonthBudgeted && {
+                  textDecoration: 'line-through',
+                  color: colors.n7,
+                },
+                styles.smallText,
+                selected && {
+                  backgroundColor: colors.p6,
+                  color: 'white',
+                },
+                (hovered || selected) && {
+                  borderRadius: 0,
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏would you mind adding a `cursor: pointer` hover state for the buttons?

Reply:
Done